### ISSUE TITLE
Add Slicer reconstructors

### ISSUE THREAD

**Issue by fedorov** [QIICR/dcmheat#1]
It would be good to add Slicer DICOM plugin based (both gdcm and dcmtk) reconstructions to the collection.

Can probably leverage https://github.com/SlicerProstate/mpReview/blob/master/mpReviewPreprocessor.py.

cc: @pieper

**Comment by fedorov** [comment]
From @jcfr:

> Here is an image to execute graphical app: https://github.com/thewtex/docker-opengl

**Comment by jcfr** [comment]
>  an image to execute graphical app: https://github.com/thewtex/docker-opengl

Credits go to @thewtex for putting the image together.

**Comment by pieper** [comment]
I made a working proof of concept test:

https://github.com/pieper/dcmheat/commit/c3c516b6a9863e89dcd4adbf125b89cc743daf83

This approach is definitely feasible and could be extended with many nice features, but it's so heavyweight I'm not sure anyone will want to use it just to convert files.

On the other hand, this same approach can be used to expose any Slicer workflow for execution as a container so that can make things more interesting.

**Comment by fedorov** [comment]
Thank you Steve! Yes, exactly - I have always been super reluctant to recommend this approach for batch conversion using Slicer. But it still may be useful to use this for testing.

Would it be possible to add a command line option to choose between dcmtk and gdcm paths?

**Comment by pieper** [comment]
Hi Andrey - I just pushed a new version with --dcmtk and --gdcm flags and
tested they work.

Agreed, for testing this method will be workable.  I'm looking forward to
seeing how the various tools perform.

-Steve

**Comment by fedorov** [comment]
@pieper thank you! Can you make a PR to the upstream repo so this script can be properly attributed?

**Comment by pieper** [comment]
Okay done!

**Comment by fedorov** [comment]
> an image to execute graphical app: https://github.com/thewtex/docker-opengl

@jcfr this didn't quite work, since I have a bit different goal - I want to run a python script via Slicer in headless mode. But I figured that out using xpra and upgrading the base image and related components to centos7. In any case, it is helpful to know about the repo you suggested.

**Comment by jcfr** [comment]
Great

> upgrading the base image

By any chance, do you have a pointer to your new Dockerfile ?

**Comment by fedorov** [comment]
> By any chance, do you have a pointer to your new Dockerfile ?

Yes: https://github.com/fedorov/docker-centos-build, see centos7 branch. Automated build at https://hub.docker.com/r/fedorov/docker-centos-build/builds/. 

I will be more than happy to contribute this back upstream to @thewtex https://github.com/thewtex/docker-centos-build to keep everything in a single location. 2 things I changed:
* I do not install `devtools` - the one used originally does not work in centos7. I figured out how to install `devtools-7`, but things seem to work without it, so I didn't install it to reduce package size
* I disabled building python, since it was causing problems with `yum`, and I didn't need it

**Comment by thewtex** [comment]
@fedorov Well done!

It would be great to have these maintained in one repository. @fedorov , I sent an invite link to https://github.com/thewtex/docker-centos-build if you want to push a `centos7` branch there.

**Comment by fedorov** [comment]
@thewtex thanks - `centos7` branch pushed
